crates/bin/src/entry.rs

```diff
@@ -185,22 +185,27 @@ pub async fn install_crates(args: Args, jobserver_client: LazyJobserverClient) -
 
 /// Return (resolvers, cargo_install_fallback)
 fn compute_resolvers(
-    input_strategies: Vec<Strategy>,
+    mut strategies: Vec<Strategy>,
     mut disable_strategies: Vec<Strategy>,
 ) -> Result<(Vec<Resolver>, bool), BinstallError> {
-    // Compute strategies
-    let mut strategies = vec![];
-
-    // Remove duplicate strategies
-    for strategy in input_strategies {
-        if strategies.len() == Strategy::COUNT {
-            // All variants of Strategy is present in strategies,
-            // there is no need to continue since all the remaining
-            // args.strategies must be present in stratetgies.
-            break;
-        }
-        if !strategies.contains(&strategy) {
-            strategies.push(strategy);
+    let dup_strategy_err =
+        BinstallError::InvalidStrategies(&"--strategies should not contain duplicate strategy");
```

Comment:
Err, what's up with the &static &static str?

Reply:
That's because `BinstallError::InvalidStrategies` takes `&'static &'static str`, though that's actually just useless since `io::Error` alone is 16B large.

Reply:
Done

Reply:
Well, turns out that `io::Error` is just 8B on 64-bit platform due to use of bitpack (storing error kind in top 16 bits of the pointer), but other variants of `BinstallError` is well larger than 8B, so it still make sense to use `&'static str`.